### PR TITLE
SD polyfill via a pseudo-isolated DOM.

### PR DESCRIPTION
**Pull request by dvoytenko**
Partial for #3742.

### CODE REVIEW

File: src/shadow-embed.js [hunk 1]
```diff
@@ -15,20 +15,98 @@
...
+  hostElement.shadowRoot = shadowRoot;
+
+  // API: https://www.w3.org/TR/shadow-dom/#the-shadowroot-interface
+
+  /** @type {!Element} */
+  shadowRoot.host = hostElement;
+
+  /** @type {function(string):?Element} */
+  shadowRoot.getElementById = function(id) {
+    const escapedId = win.CSS && win.CSS.escape && win.CSS.escape(id) || id;
```
Comment (cramforce) [inline src/shadow-embed.js]: Lets factor this out.

File: src/shadow-embed.js [hunk 2]
```diff
@@ -47,3 +125,91 @@ export function createShadowEmbedRoot(hostElement, extensionIds) {
...
+ * @param {string} css
+ * @return {string}
+ */
+export function transformShadowCss(css) {
+  if (isShadowDomSupported()) {
+    return css;
+  }
+
+  // Replace html/body with amp-html/amp-body.
+  const res = css.replace(/(html|body)/g, function(match, name, pos) {
```
Comment (cramforce) [inline src/shadow-embed.js]: Should this put `amp-body` before every selector?

Comment (dvoytenko) [inline src/shadow-embed.js]: Yeah, we could totally do that. I guess there are degrees of isolation we can add. I thought this is an ok start and add more in follow up PRs. I'm currently investigating `<style scoped>` and `:scope` that could potentially help in this.

Comment (cramforce) [inline src/shadow-embed.js]: I think `scoped` is deprecated and `:scope` is `querySelector` only.

If we want to make this for real, we will need to add the style scoping I think

It should probably be `selector` becomes `#root123 selector`

File: src/shadow-embed.js (hunk 1 above)
Comment (dvoytenko) [inline src/shadow-embed.js]: Done

File: src/shadow-embed.js (hunk 2 above)
Comment (dvoytenko) [inline src/shadow-embed.js]: Agree. But ok to do it in a follow up PR?

**Comment by dvoytenko** [comment]
@cramforce all requests are addressed. PTAL.

**Comment by cramforce** [comment]
LGTM

**Pull request merged**